app/code/Magento/ConfigurableProduct/Model/Product/Type/Collection/SalableProcessor.php

```diff
@@ -38,19 +38,20 @@ public function __construct(StatusFactory $stockStatusFactory)
      * when filter already added and therefore may break the collection.
      *
      * @param Collection $collection
+     * @param bool $isFilterInStock
      * @return Collection
      */
-    public function process(Collection $collection)
+    public function process(Collection $collection, $isFilterInStock = true)
     {
         $collection->addAttributeToFilter(
             ProductInterface::STATUS,
             Status::STATUS_ENABLED
         );
 
         $stockFlag = 'has_stock_status_filter';
-        if (!$collection->hasFlag($stockFlag)) {
+        if (!$collection->hasFlag($stockFlag) || !$collection->getFlag($stockFlag)) {
```

Comment:
I see no advantage in adding ` || !$collection->getFlag($stockFlag)`, since the value of this flag is irrelevant as long it is set, because its presence should be only used to mark the collection has already added the stock data to collection, to prevent query errors if \Magento\CatalogInventory\Model\ResourceModel\Stock\Status::addStockDataToCollection is called twice with the same collection before the collection is loaded. Also, this processor logic is very similar to 
\Magento\CatalogInventory\Helper\Stock::addIsInStockFilterToCollection, if we preserved this here, we'd have to adapt also this method.

Reply:
`if (!$collection->hasFlag($stockFlag)) {`